pkg/nodeinfo/fetch_node_state_vsql.go

```diff
@@ -0,0 +1,118 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package nodeinfo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+)
+
+func (v *VSQL) FetchNodeState(ctx context.Context) (*NodeInfo, error) {
+	sql := v.buildFetchNodeStateQuery()
+	stdout, err := v.queryNodeStatus(ctx, sql)
+	if err != nil {
+		// Skip parsing that happens next
+		return nil, nil
```

Comment:
Should we return the error?
```suggestion
		return nil, err
```

Reply:
> Should we return the error?

Do we want that? I thought we were just going to move on to the next function without setting those podfact fields.

Reply:
Oh, I see that was copied from before. I would still return the error in the package and let the caller decide if they want to ignore the error and continue collecting facts.